services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts

```diff
@@ -74,17 +85,17 @@ export class TransactionJobProcessorsService {
         input[key] === ''
       ) {
         const errorText = `Property ${key} is undefined`;
-        await this.createTransaction({
-          amount: input.transactionAmount,
-          registration: registration,
+        await this.processTransactionResult({
+          jobInput: input,
+          calculatedTranserAmount: input.transactionAmount, // TODO: STORE THE CALCULATED AMOUNT HERE AS THIS IS USED ON RETRY AND SHOWN IN PORTAL
           financialServiceProviderId: financialServiceProvider.id,
-          programId: input.programId,
-          paymentNumber: input.paymentNumber,
-          userId: input.userId,
+          registration,
+          oldRegistration,
+          isRetry: input.isRetry,
           status: StatusEnum.error,
-          errorMessage: errorText,
+          errorText,
         });
-        throw new Error(errorText);
+        return;
```

Comment:
Return instead of error, to not pollute our logs

Reply:
What does "processTransactionResult" conceptually mean?

jobInput: I do not like passing in "big object parameters) into functions and then having functions internally figure out what they need. I know this happens a lot now throughout our code, for example with PaPaymentDetails "blobs" which are inobvious about what they conceptually are/contain, and get passed around throughout a service. IMO every function should have a "clean" interface, only receiving what it needs and passing back what the caller needs.

Yes, good point about storing the transactionAmount in the failed transaction. However, at this point it is not a calculated amount yet, since that happens in the IntersolveVisa service. Unless you mean that it is calculated already with the multiplier. If that is the case, check where that happens. If that is when creating the job, then IMO we should not be talking about "calculated" amount here at all.
So what is "calculatedTrans(f)Amount" conceptually?

Yes, you have a point about not rethrowing the error. There could be a misconfiguration in the program, but that is not "our problem".

Reply:
Renamed in this [commit](https://github.com/global-121/121-platform/pull/5617/commits/e21c38f551a6a900453abc914b3e9fc6b024b641)